components/support/sql/src/conn_ext.rs

```diff
@@ -221,16 +221,15 @@ pub trait ConnExt {
         })
     }
 
-    fn unchecked_transaction(&self) -> SqlResult<UncheckedTransaction> {
-        UncheckedTransaction::new(self.conn(), TransactionBehavior::Deferred)
-    }
-
-    fn time_chunked_transaction(
-        &self,
-        commit_after: Duration,
-    ) -> SqlResult<TimeChunkedTransaction> {
-        TimeChunkedTransaction::new(self.conn(), TransactionBehavior::Deferred, commit_after)
-    }
+    // I think we should either remove this (which would mean copying it into logins)
```

Comment:
I think just having the rule is probably sufficient, TBH. It already has `unchecked_`in the name, which sort of discourages its use.

Reply:
> It already has `unchecked_`in the name, which sort of discourages its use.

Right, but that's the opposite of the intent - `unchecked_transaction` is our *preferred* (and in practice, the only available due to not passing `mut` everywhere) transaction mechanism in our toolbelt.

Reply:
I've left `unchecked_transaction` available for places and we'll just need to enforce a rule that it shouldn't be used (although it actually doesn't matter if it is used to perform reads)